projects/ngx-cookie-service-ssr/src/lib/ssr-cookie.service.ts

```diff
@@ -54,6 +56,71 @@ export class SsrCookieService {
     }
   }
 
+  /**
+   * Converts the provided cookie string to a key-value representation.
+   *
+   * @param cookieString - A concatenated string of cookies
+   * @returns Map - Key-value pairs of the provided cookies
+   *
+   * @author: Blake Ballard (blakeoxx)
+   * @since: 18.1.0
+   */
+  static cookieStringToMap(cookieString: string): Map<string, string> {
+    const cookies = new Map<string, string>;
+
+    if (cookieString?.length < 1) {
+      return cookies;
+    }
+
+    cookieString.split(';').forEach((currentCookie) => {
+      let [cookieName, cookieValue] = currentCookie.split('=');
+
+      // Remove any extra spaces from the beginning of cookie names. These are a side effect of browser/express cookie concatenation
+      cookieName = cookieName.replace(/^ +/, '');
```

Comment:
IMO, trimStart might be more straightforward.

Reply:
I used this to keep consistent with previous code, but agreed, happy to change to `trimStart`.

See https://github.com/stevermeister/ngx-cookie-service/pull/320/files#diff-4bd5ade27ff959a3d837687c0ae49f2f05cdabb3b655a61cd4c0a03497171c10L109